src/frontend/components/DashboardVideoLiveControlPane/widgets/DashboardVideoLiveWidgetThumbnail/ThumbnailManager/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { Box, Text } from 'grommet';
+import React from 'react';
+import { defineMessages, useIntl } from 'react-intl';
+
+import { ThumbnailDisplayer } from 'components/DashboardVideoLiveControlPane/widgets/DashboardVideoLiveWidgetThumbnail/ThumbnailDisplayer';
+import {
+  UploadManagerState,
+  UploadManagerStatus,
+} from 'components/UploadManager';
+import { Thumbnail, uploadState } from 'types/tracks';
+
+const messages = defineMessages({
+  noThumbnailUploaded: {
+    defaultMessage:
+      'You have no thumbnail uploaded, the default thumbnail will be used.',
+    description: 'A text displayed when there is no uploaded thumbnail.',
+    id: 'components.ThumbnailManager.noThumbnailUploaded',
+  },
+  thumbnailUploaded: {
+    defaultMessage: 'Your image is being uploaded.',
+    description: 'A text displayed when the thumbnail is being uploaded.',
+    id: 'components.ThumbnailManager.thumbnailUploaded',
+  },
+  thumbnailProcessed: {
+    defaultMessage: 'Your image is being processed.',
+    description: 'A text displayed when the thumbnail is being processed.',
+    id: 'components.ThumbnailManager.thumbnailProcessed',
+  },
+});
+
+interface ThumbnailManagerProps {
+  thumbnail: Thumbnail;
+  uploadManagerState: UploadManagerState;
+}
+
+export const ThumbnailManager = ({
+  thumbnail,
+  uploadManagerState,
+}: ThumbnailManagerProps) => {
+  const intl = useIntl();
+  const thumbnailState = thumbnail ? thumbnail.upload_state : uploadState.READY;
+
+  let textMessage = null;
+
+  if (thumbnailState === uploadState.ERROR) {
```

Comment:
It looks like you can externalize these `if else if` from this component (like in a dedicated file).

Reply:
mm the point of the ThumbnailManager was alredy to externalize the logic you are talking about